### PR TITLE
Updating the OpenAPI Guide with latest SmallRye features

### PR DESCRIPTION
**Pull request by phillip-kruger**
Small changes to the OpenAPI Guide to include some new features from SmallRye OpenAPI.
- Overriding of the version (3.0.3)
- Creating the global info using configuration
- Auto-generation of the Operation Id

Signed-off-by: Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 1]
```diff
@@ -302,6 +302,27 @@ public class ExampleApiApplication extends Application {
 }
 ----
 
+Another option, that is a feature provided by SmallRye and not part of the specification, is to use configuration to add this global API information.
+This way you do not need to a JAX-RS `Application` class, and you can name the API different per environment.
```
Comment (machi1990) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:306]:
```suggestion
This way, you do not need to have a JAX-RS `Application` class, and you can name the API different per environment.
```

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 2]
```diff
@@ -385,6 +406,36 @@ Alternative paths are:
...
+[source, properties]
+----
+mp.openapi.extensions.smallrye.operationIdStrategy=METHOD
+----
+
+Now you do not need to use the `@Operation` annotation. While generating the document, the method name will be used for the Operation Id.
+
+The strategies available for generating the Operation Id is:
+
+* METHOD - Use the method name.
```
Comment (machi1990) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:435]:
```suggestion
* `METHOD` - Use the method name.
```

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 3]
```diff
@@ -385,6 +406,36 @@ Alternative paths are:
...
+* CLASS_METHOD - Use the class name (without the package) plus the method.
```
Comment (machi1990) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:436]:
```suggestion
* `CLASS_METHOD` - Use the class name (without the package) plus the method.
```

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 4]
```diff
@@ -385,6 +406,36 @@ Alternative paths are:
...
+* PACKAGE_CLASS_METHOD - Use the class name plus the method name.
```
Comment (machi1990) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:437]:
```suggestion
* `PACKAGE_CLASS_METHOD` - Use the class name plus the method name.
```

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 5]
```diff
@@ -385,6 +406,36 @@ Alternative paths are:
...
+The Operation Id is typically used for the method name in the client stub. In SmallRye, you can auto-generate this Operation Id by using the following configuration:
+
+[source, properties]
+----
+mp.openapi.extensions.smallrye.operationIdStrategy=METHOD
+----
+
+Now you do not need to use the `@Operation` annotation. While generating the document, the method name will be used for the Operation Id.
+
+The strategies available for generating the Operation Id is:
```
Comment (machi1990) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:433]:
```suggestion
The strategies available for generating the Operation Id are:
```

Comment (machi1990) [reply docs/src/main/asciidoc/openapi-swaggerui.adoc:433]: Also wondering if this should be a table, just like [Database kind to JDBC driver mapping](https://github.com/quarkusio/quarkus/blob/df5ee4e1974b280eba81b633c48b0a7d08c37f72/docs/src/main/asciidoc/datasource.adoc#configure-the-jdbc-connection) in the `datasource.adoc` guide

**Review by machi1990 — DISMISSED** [review]

File: docs/src/main/asciidoc/openapi-swaggerui.adoc (hunk 5 above)
Comment (phillip-kruger) [reply docs/src/main/asciidoc/openapi-swaggerui.adoc]: Agree. Made it a table

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 6]
```diff
@@ -302,6 +302,27 @@ public class ExampleApiApplication extends Application {
 }
 ----
 
+Another option, that is a feature provided by SmallRye and not part of the specification, is to use configuration to add this global API information.
+This way, you do not need to have a JAX-RS `Application` class, and you can name the API different per environment.
```
Comment (gsmet) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:306]:
```suggestion
This way, you do not need to have a JAX-RS `Application` class, and you can name the API differently per environment.
```

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 7]
```diff
@@ -302,6 +302,27 @@ public class ExampleApiApplication extends Application {
...
+
+Example, add the following to your `application.properties`:
```
Comment (gsmet) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:308]:
```suggestion
For example, add the following to your `application.properties`:
```

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 8]
```diff
@@ -385,6 +406,44 @@ Alternative paths are:
 Live reload of static OpenAPI document is supported during development. A modification to your OpenAPI document will be picked up on fly by Quarkus.
 ====
 
+== Changing the openapi version
```
Comment (gsmet) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:409]: Should it be OpenAPI? And below too?

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 9]
```diff
@@ -385,6 +406,44 @@ Alternative paths are:
...
+[source, properties]
+----
+mp.openapi.extensions.smallrye.openapi=3.0.2
+----
+
+This might be useful if your API go through a Gateway that needs a certain version.
+
+== Auto-generation of OperationId
+
+The https://swagger.io/docs/specification/paths-and-operations/[Operation Id] can be set using the `@Operation` annotation, and is in many cases useful when using a tool to generate a client stub from the schema.
```
Comment (gsmet) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:423]: Sorry for the nitpicking but you write `OperationId` in the title and then `Operation Id` here and twice below. Which one is it? Better be consistent.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added some minor comments.

**Comment by phillip-kruger** [comment]
Thanks @gsmet and @machi1990 . I made all changes as suggested and squashed.

**Review by gsmet — APPROVED** [review]

**Review by machi1990 — APPROVED** [review]

File: docs/src/main/asciidoc/openapi-swaggerui.adoc [hunk 10]
```diff
@@ -302,6 +302,27 @@ public class ExampleApiApplication extends Application {
...
+mp.openapi.extensions.smallrye.info.description=Just an example service
+mp.openapi.extensions.smallrye.info.termsOfService=Your terms here
+mp.openapi.extensions.smallrye.info.contact.email=techsupport@example.com
+mp.openapi.extensions.smallrye.info.contact.name=Example API Support
+mp.openapi.extensions.smallrye.info.contact.url=http://exampleurl.com/contact
+mp.openapi.extensions.smallrye.info.license.name=Apache 2.0
+mp.openapi.extensions.smallrye.info.license.url=http://www.apache.org/licenses/LICENSE-2.0.html
+----
+
+This will give you simular information as the `@OpenAPIDefinition` example above.
```
Comment (Postremus) [inline docs/src/main/asciidoc/openapi-swaggerui.adoc:325]: should be 'similar'

Comment (gsmet) [reply docs/src/main/asciidoc/openapi-swaggerui.adoc]: Good catch!

**Pull request merged**